### PR TITLE
fix: refactor code->runtime methods so that CtAnnotation#getValueOfObject does not overflow

### PR DESCRIPTION
**Pull request by seintur**
Failing test case for CtAnnotationImpl#getValueAsObject for https://github.com/INRIA/spoon/issues/2858

Don't have a fix yet.

### CODE REVIEW

**Comment by monperrus** [comment]
Thanks Lionel. I'm looking at it.

**Comment by monperrus** [comment]
the reason is that the partial evaluator does not know how to evaluate `\"a+\"+Integer.SIZE`

What's the best behavior to your opinion?

- null
- Object()
- exception?

**Comment by seintur** [comment]
I guess an exception that would mention that the evaluation is unsupported would be best.
Thanks.

**Comment by monperrus** [comment]
@oscarlvp there is a gift for you here :-)

File: src/test/java/spoon/test/annotation/AnnotationTest.java [hunk 1]
```diff
@@ -1547,4 +1547,17 @@ public void testAnnotationArray() throws Exception {
 		assertTrue(shadowAnnotationOne.getValue("role") instanceof CtLiteral); // should be CtLiteral
 		assertEquals(annotationOne.getValue("role"), shadowAnnotationOne.getValue("role")); // should pass
 	}
+
+	@Test(expected = SpoonException.class)
+	public void testGetValueAsObject() {
+		// contract: when one cannt evaluate an expression, throw an exception
```
Comment (seintur) [inline src/test/java/spoon/test/annotation/AnnotationTest.java]: cannt -> cannot

Comment (monperrus) [reply src/test/java/spoon/test/annotation/AnnotationTest.java]: Thanks, but already merged. Could you do a live pull-request with web-based editing?

**Pull request merged**